default.nix

```
@@ -1,116 +1,148 @@
-      pname = "agda-finset";
-      version = "0.9";
-      src = fetchFromGitHub {
-        repo = "agda-finset";
-        owner = "input-output-hk";
-        rev = "939b2578f4f8cb4f02928b30edfc787cabeba171";
-        sha256 = "2bUMmUikzNRaEFVkH+Y8ypnNF65d/LNKei6fSJ982AY=";
-      };
-      meta = {};
-      libraryFile = "Finset.agda-lib";
-      everythingFile = "src/README.agda";
-      buildInputs = [ agda-stdlib ];
+  agdaStdlib = customAgda.agdaPackages.standard-library.overrideAttrs (oldAttrs: {
+    version = "1.7";
+    src = customAgda.fetchFromGitHub {
+      repo = "agda-stdlib";
+      owner = "input-output-hk";
+      rev = "f8fdb925c74e8d3b0c88e2a5520bc11e606d34c6";
+      sha256 = "BoK/IZsOn8gnUolI8DOZa6IOoXF8E95s2e8vZyUpMZs=";
     };
+  });
 
-    agda-stdlib-meta = agdaPackages.mkDerivation {
-      pname = "agda-stdlib-meta";
-      version = "0.1";
-      src = fetchFromGitHub {
-        repo = "stdlib-meta";
-        owner = "omelkonian";
-        rev = "dadb6a468b9cdc47442b48a47b848f8e8fbffda7";
-        sha256 = "YkUtM5Gos6xd7ZsZPqcuVy6DZqNA7n/exPfQngir+y0=";
-      };
-      patches = [ ./stdlib-meta-update-imports.patch ];
-      meta = {};
-      libraryFile = "stdlib-meta.agda-lib";
-      everythingFile = "Main.agda";
-      buildInputs = [ agda-stdlib ];
+  agdaFinset = customAgda.agdaPackages.mkDerivation {
+    pname = "agda-finset";
+    version = "0.9";
+    src = customAgda.fetchFromGitHub {
+      repo = "agda-finset";
+      owner = "input-output-hk";
+      rev = "939b2578f4f8cb4f02928b30edfc787cabeba171";
+      sha256 = "2bUMmUikzNRaEFVkH+Y8ypnNF65d/LNKei6fSJ982AY=";
     };
+    meta = { };
+    libraryFile = "Finset.agda-lib";
+    everythingFile = "src/README.agda";
+    buildInputs = [ agdaStdlib ];
+  };
 
-    deps = [ agda-stdlib agda-finset agda-stdlib-meta ];
-  in {
-    agda = agda.withPackages { pkgs = deps; ghc = nixpkgs.ghc; };
-    ledger = agdaPackages.mkDerivation {
-      pname = "Agda-ledger";
-      version = "0.1";
-      src = ./src;
-      meta = {};
-      buildInputs = deps;
-      postInstall = "cp -r latex $out";
-      extraExtensions = [ "hs" "cabal" ];
+  agdaStdlibMeta = customAgda.agdaPackages.mkDerivation {
+    pname = "agda-stdlib-meta";
+    version = "0.1";
+    src = customAgda.fetchFromGitHub {
+      repo = "stdlib-meta";
+      owner = "omelkonian";
+      rev = "dadb6a468b9cdc47442b48a47b848f8e8fbffda7";
+      sha256 = "YkUtM5Gos6xd7ZsZPqcuVy6DZqNA7n/exPfQngir+y0=";
     };
+    patches = [ ./stdlib-meta-update-imports.patch ];
+    meta = { };
+    libraryFile = "stdlib-meta.agda-lib";
+    everythingFile = "Main.agda";
+    buildInputs = [ agdaStdlib ];
   };
-  agda-ledger = agdaDeps.ledger;
-  agdaWithPkgs = agdaDeps.agda;
 
-  ledger-hs-src = stdenv.mkDerivation {
-    pname = "Agda-ledger-hs-src";
+  deps = [ agdaStdlib agdaFinset agdaStdlibMeta ];
+  agdaWithPkgs = customAgda.agda.withPackages { pkgs = deps; ghc = pkgs.ghc; };
+
+in
+rec {
+
```

We need to make agda itself available as well
```suggestion
  agda = agdaWithPkgs;
```